apis/apps/v1/cluster_types.go

```diff
@@ -0,0 +1,866 @@
+	// InstanceTemplate allows for specifying these unique configurations per instance.
+	// Each instance's name is constructed using the pattern: $(component.name)-$(template.name)-$(ordinal),
+	// starting with an ordinal of 0.
+	// It is crucial to maintain unique names for each InstanceTemplate to avoid conflicts.
+	//
+	// The sum of replicas across all InstanceTemplates should not exceed the total number of replicas specified for the Component.
+	// Any remaining replicas will be generated using the default template and will follow the default naming rules.
+	//
+	// +optional
+	// +patchMergeKey=name
+	// +patchStrategy=merge,retainKeys
+	// +listType=map
+	// +listMapKey=name
+	Instances []InstanceTemplate `json:"instances,omitempty" patchStrategy:"merge,retainKeys" patchMergeKey:"name"`
+
+	// Specifies the names of instances to be transitioned to offline status.
+	//
+	// Marking an instance as offline results in the following:
+	//
+	// 1. The associated Pod is stopped, and its PersistentVolumeClaim (PVC) is retained for potential
+	//    future reuse or data recovery, but it is no longer actively used.
+	// 2. The ordinal number assigned to this instance is preserved, ensuring it remains unique
+	//    and avoiding conflicts with new instances.
+	//
+	// Setting instances to offline allows for a controlled scale-in process, preserving their data and maintaining
+	// ordinal consistency within the Cluster.
+	// Note that offline instances and their associated resources, such as PVCs, are not automatically deleted.
+	// The administrator must manually manage the cleanup and removal of these resources when they are no longer needed.
+	//
+	// +optional
+	OfflineInstances []string `json:"offlineInstances,omitempty"`
+
+	// Determines whether metrics exporter information is annotated on the Component's headless Service.
+	//
+	// If set to true, the following annotations will not be patched into the Service:
+	//
+	// - "monitor.kubeblocks.io/path"
+	// - "monitor.kubeblocks.io/port"
+	// - "monitor.kubeblocks.io/scheme"
+	//
+	// These annotations allow the Prometheus installed by KubeBlocks to discover and scrape metrics from the exporter.
+	//
+	// +optional
+	DisableExporter *bool `json:"disableExporter,omitempty"`
+}
+
+type ServiceRef struct {
+	// Specifies the identifier of the service reference declaration.
+	// It corresponds to the serviceRefDeclaration name defined in either:
+	//
+	// - `componentDefinition.spec.serviceRefDeclarations[*].name`
+	// - `clusterDefinition.spec.componentDefs[*].serviceRefDeclarations[*].name` (deprecated)
+	//
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// Specifies the namespace of the referenced Cluster or the namespace of the referenced ServiceDescriptor object.
+	// If not provided, the referenced Cluster and ServiceDescriptor will be searched in the namespace of the current
+	// Cluster by default.
+	//
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
```

Comment:
as discussed, references a Cluster in a different namespace is forbidden because of the security consideration (referencing a Secret in another ns). so may be we can delete this field